Automattic/sensei
# Add core supports for custom blocks

Fixes #5536 

### Changes proposed in this Pull Request

* Organises learning mode blocks into `block-name.block.json` files.
* Shares block meta-data between JS and PHP
* Adds some core block 'supports' features to LM blocks for easier editing/customisation.
* Uses `get_block_wrapper_attributes` to add customisations added in the editor to the PHP render function for the frontend.

### Testing instructions
* Go to the editor for a lesson
* Select any LM block (There a couple of them, typing 'theme' or 'course' in the block inserter will bring them up and they all have the Sensei logo as icons so it's easy to identify, you can also find a list of them in `assets/course-theme/blocks/...`).
* After inserting a block, make sure it is selectable.
* Select a block and look at the block settings for that block. Notice the settings panel has UI controls for colour, typography etc. 
* Try changing some of them and then save.
* Refresh the page to see the changes were persisted.
* View the lesson in the frontend, ensure your changes from the editor reflect.
